src/main/java/service/TransactionServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package service;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import model.FruitTransaction;
+import service.transaction.TransactionHandler;
+import service.transaction.TransactionStrategy;
+import service.transaction.TransactionStrategyImpl;
+
+public class TransactionServiceImpl implements TransactionService {
```

Comment:
This service needs to be broken down into several. You should not have different tasks in one service. Again, these are all solid principles